jni/src/faiss_wrapper.cpp

```diff
@@ -192,9 +192,8 @@ jobjectArray knn_jni::faiss_wrapper::QueryIndex(knn_jni::JNIUtilInterface * jniU
         throw std::runtime_error("Invalid pointer to index");
     }
 
-    int dim	= jniUtil->GetJavaFloatArrayLength(env, queryVectorJ);
-    std::vector<float> dis(kJ * dim);
-    std::vector<faiss::Index::idx_t> ids(kJ * dim);
+    std::vector<float> dis(kJ);
+    std::vector<faiss::Index::idx_t> ids(kJ);
```

Comment:
add a comment on top of this change what it is doing and why it is required.

Reply:
Added the comment.